shared/chat/conversation/info-panel/add-to-channel/index.new.tsx

```diff
@@ -0,0 +1,201 @@
+
+  const [toAdd, setToAdd] = React.useState<Set<string>>(new Set())
+  const [filter, setFilter] = React.useState('')
+  const fLower = filter.toLowerCase()
+
+  const {channelname, teamname, teamID} = Container.useSelector(s =>
+    ChatConstants.getMeta(s, conversationIDKey)
+  )
+  const participants = Container.useSelector(s => ChatConstants.getParticipantInfo(s, conversationIDKey)).all
+  const teamDetails = Container.useSelector(s => TeamConstants.getTeamDetails(s, teamID))
+  const allMembers = sortMembers(teamDetails.members)
+  const membersFiltered = allMembers.filter(
+    m => m.username.toLowerCase().includes(fLower) || m.fullName.toLowerCase().includes(fLower)
+  )
+
+  useTeamDetailsSubscribe(teamID)
+
+  const onClose = () => dispatch(nav.safeNavigateUpPayload())
+  const onAdd = () => {
+    dispatch(Chat2Gen.createAddUsersToChannel({conversationIDKey, usernames: [...toAdd]}))
+    onClose()
+  }
+
+  const loading = !allMembers.length
+
+  return (
+    <Kb.Modal
+      header={{
+        hideBorder: Styles.isMobile,
+        leftButton: Styles.isMobile ? (
+          <Kb.Text type="BodyBigLink" onClick={onClose}>
+            Cancel
+          </Kb.Text>
+        ) : (
+          undefined
+        ),
+        rightButton: Styles.isMobile && toAdd.size && (
+          <Kb.Text type="BodyBigLink" onClick={onAdd}>
+            Add
+          </Kb.Text>
+        ),
+        title: title({channelname, teamname}),
+      }}
+      footer={
+        Styles.isMobile
+          ? undefined
+          : {
+              content: (
+                <Kb.Box2 direction="horizontal" gap="tiny" fullWidth={true}>
+                  <Kb.Button
+                    type="Dim"
+                    label="Cancel"
+                    onClick={onClose}
+                    style={Styles.globalStyles.flexOne}
+                  />
+                  <Kb.Button
+                    label={toAdd.size ? `Add ${toAdd.size} ${pluralize('member', toAdd.size)}` : 'Add...'}
+                    onClick={onAdd}
+                    disabled={!toAdd.size}
+                    style={Styles.globalStyles.flexOne}
+                  />
+                </Kb.Box2>
+              ),
+            }
+      }
+      onClose={onClose}
+      allowOverflow={true}
+    >
+      <Kb.SearchFilter
+        onChange={text => setFilter(text)}
+        size="full-width"
+        placeholderText={
+          loading ? 'Loading...' : `Search ${allMembers.length} ${pluralize('member', allMembers.length)}`
+        }
+        style={styles.filterInput}
+      />
+      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.listContainer}>
+        <Kb.List2
+          items={membersFiltered}
+          renderItem={(idx, item) => {
+            const alreadyIn = participants.includes(item.username)
+            const onCheck = () => {
+              if (toAdd.has(item.username)) {
+                toAdd.delete(item.username)
+                setToAdd(new Set(toAdd))
```

Comment:
Why does this need to be a new `Set`?

Reply:
To make sure the reference changes and react triggers an update